doc/RecordingSearch.xml

```diff
@@ -1320,6 +1320,193 @@ http://www.onvif.org/ver10/tptz/ZoomSpaces/PositionGenericSpace
+          <listitem>
+            <para role="access">READ_MEDIA</para>
+          </listitem>
+        </varlistentry>
+      </variablelist>
+    </section>
+    
+    <section>
+      <title>GetNLSearchResults</title>
+      <para>GetNLSearchResults acquires the results from a natural language search session previously initiated by a SearchImageByNL operation. The response shall not include results already returned in previous requests for the same session.</para>
+      <variablelist role="op">
+        <varlistentry>
+          <term>request</term>
+          <listitem>
+            <para role="param">SearchToken [tt:JobToken]</para>
+            <para role="text">Specifies the search session.</para>
+            <para role="param">MinResults - optional [xs:int]</para>
+            <para role="text">Specifies the minimum number of results that should be returned.</para>
+            <para role="param">MaxResults – optional [xs:int]</para>
+            <para role="text">Specifies the maximum number of results to return.</para>
+            <para role="param">WaitTime – optional [xs:duration]</para>
+            <para role="text">Defines the maximum time to block, waiting for results.</para>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <term>response</term>
+          <listitem>
+            <para role="param">ResultList [tt:FindImageResultList]</para>
+            <para role="text">A structure containing the search results.</para>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <term>faults</term>
+          <listitem>
+            <para role="param">env:Sender - ter:InvalidArgVal - ter:InvalidToken</para>
+            <para role="text">The search token is invalid.</para>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <term>access class</term>
+          <listitem>
+            <para role="access">READ_MEDIA</para>
+          </listitem>
+        </varlistentry>
+      </variablelist>
+    </section>
+    
+    <section>
+      <title>SearchImageByImage</title>
+      <para>SearchImageByImage starts a search session, looking for video records based on a provided image. Results from the search are acquired using the GetImageSearchResults request, specifying the search token returned from this request.</para>
+      <variablelist role="op">
+        <varlistentry>
+          <term>request</term>
+          <listitem>
+            <para role="param">StartPoint [xs:dateTime]</para>
+            <para role="text">The point of time where the search will start.</para>
+            <para role="param">EndPoint [xs:dateTime]</para>
+            <para role="text">The point of time where the search will stop.</para>
+            <para role="param">RecordingToken - optional [tt:RecordingReference]</para>
+            <para role="text">Token for the recording to search.</para>
+            <para role="param">TargetImageURI [xs:anyURI]</para>
```

Comment:
add a search parameter to accept external image from client in addition to NPL Target image URI. Either client can use NPLTargetImageURI or External Image from client for image search feature.